packages/flutter/lib/src/material/search_bar_theme.dart

```diff
@@ -0,0 +1,280 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material_state.dart';
+import 'theme.dart';
+
+// Examples can assume:
+// late BuildContext context;
+
+/// Defines default property values for descendant [SearchBar] widgets.
+///
+/// Descendant widgets obtain the current [SearchBarThemeData] object using
+/// `SearchBarTheme.of(context)`. Instances of [SearchBarThemeData] can be customized
+/// with [SearchBarThemeData.copyWith].
+///
+/// Typically a [SearchBarThemeData] is specified as part of the overall [Theme]
+/// with [ThemeData.searchBarTheme].
+///
+/// All [SearchBarThemeData] properties are `null` by default. When null, the
+/// [SearchBar] will use the values from [ThemeData] if they exist, otherwise it
+/// will provide its own defaults based on the overall [Theme]'s colorScheme.
+/// See the individual [SearchBar] properties for details.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class SearchBarThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.searchBarTheme].
+  const SearchBarThemeData({
+    this.elevation,
+    this.backgroundColor,
+    this.shadowColor,
+    this.surfaceTintColor,
+    this.overlayColor,
+    this.side,
+    this.shape,
+    this.padding,
+    this.textStyle,
+    this.hintStyle,
+    this.constraints,
+  });
+
+  /// The elevation of the search bar's [Material].
```

Comment:
```suggestion
  /// Overrides the default value of [SearchBar.elevation].
```
Apply this change to all of the properties. This should obviate having the extra explanatory line for each property. By not restating the meaning of the property here - it's only specified int the component's API doc - we can eliminate the risk of inconsistency.